File: logica_programacao/Collatz.java
public class Collatz {

	public static void main(String[] args) {
		long maior = 1;
		long numSeqMaior = 0;

		for (long n = 1; n < 1000000; n++) {
			long val = n;
			long numseq=1;
			while (val > 1) {
				if (val % 2 == 0) {
					val /= 2;
				} else {
					val = 3 * val + 1;
				}
				numseq++;
			}
			if (numseq > numSeqMaior) {
				maior = n;
				numSeqMaior = numseq;
			}
		}
		System.out.println(maior + " " + numSeqMaior);
	}

}

File: logica_programacao/Collatz.py
maior = 1
numSeqMaior = 0
n = 1
while (n < 1000000):
	val = n
	numseq = 1
	while (val > 1):
		if (val % 2 == 0):
			val = val / 2
		else:
			val = 3 * val + 1
		numseq = numseq + 1

	if (numseq > numSeqMaior):
		maior = n
		numSeqMaior = numseq

	n = n+1

print maior
print numSeqMaior 
